File: boj/problem2581/Main.java
package problem2581;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {
    public static String solve(int M, int N) {
        int answer = -1;
        int[] arr = new int[N + 1];
        arr[1] = -1;
        int sum = 0;
        int minNum = Integer.MAX_VALUE;

        for (int i = M; i < N + 1; i++) {
            if (arr[i] == -1)
                continue;

            int divisible = 0;
            for (int j = 1; j < i / 2 + 1; j++) {
                if (i % j == 0) {
                    divisible = j;
                }
            }

            if (divisible == 1) {
                sum += i;
                minNum = Math.min(i, minNum);
                for (int j = i + i; j < N + 1; j += i) {
                    arr[j] = -1;
                }
            }
        }

        if (sum == 0 && minNum == Integer.MAX_VALUE)
            return Integer.toString(answer);

        return sum + "\n" + minNum;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int M = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        bw.write(solve(M, N));
        bw.flush();
    }
}

File: boj/problem2581/main.py
def solve(M: int, N: int) -> str:
    primes = [0] * (N + 1)
    primes[1] = -1

    total = 0
    min_num = N + 1
    for i in range(M, N + 1):
        if primes[i] == -1:
            continue

        divisible = 0
        for j in range(1, i // 2 + 1):
            if i % j == 0:
                divisible = j

        if divisible == 1:
            total += i
            min_num = min(i, min_num)
            for j in range(i + i, N + 1, i):
                primes[i] = -1

    if total == 0 and min_num == N + 1:
        return str(-1)

    return str(total) + "\n" + str(min_num)


def main():
    M = int(input())
    N = int(input())

    print(solve(M, N))


if __name__ == "__main__":
    main()
